packages/Webkul/Theme/src/Repositories/ThemeCustomizationRepository.php

```diff
@@ -19,24 +19,54 @@ public function model(): string
         return ThemeCustomization::class;
     }
 
+    /**
+     * Update the specified theme
+     *
+     * @param  int  $id
+     */
+    public function update(array $data, $id): ThemeCustomization
+    {
+        $locale = core()->getRequestedLocaleCode();
+
+        if ($data['type'] == 'static_content') {
+            $data[$locale]['options']['html'] = preg_replace('/<script\b[^>]*>(.*?)<\/script>/is', '', $data[$locale]['options']['html']);
+            $data[$locale]['options']['css'] = preg_replace('/<script\b[^>]*>(.*?)<\/script>/is', '', $data[$locale]['options']['css']);
+        }
+
+        $data['status'] = request()->input('status') == 'on';
+
+        if (in_array($data['type'], ['image_carousel', 'services_content'])) {
+            unset($data[$locale]['options']);
+        }
+
+        $theme = parent::update($data, $id);
+
+        if (in_array($data['type'], ['image_carousel', 'services_content'])) {
+            $this->uploadImage(request()->all(), $theme);
+        }
+
+        return $theme;
+    }
+
     /**
      * Upload images
      *
```

Comment:
Need to define type of the variable $theme

Reply:
done

Reply:
@suraj-webkul 

[In this code](https://github.com/bagisto/bagisto/blob/322e29c3cb1c986af5aa71c98f850cd3ca02632b/packages/Webkul/Theme/src/Repositories/ThemeCustomizationRepository.php#L94-L96), title key is missing. Without it, there is no title key in the database and blank in UI.

```
$options['images'][] = [
    'image' => 'storage/' . $path,
    'link'  => $image['link'],
    'title' => $image['title'], // <-- missing key
];
```

### Video

https://github.com/bagisto/bagisto/assets/12383757/ddce9b14-cb9a-4fab-a724-8c66aa34f119